lib/pf/pfcmd/checkup.pm

```diff
@@ -1082,6 +1084,19 @@ sub apache_filter_rules {
     }
 }
 
+=item db_version
+
+Make sure the database matches the current version
+Return the current version if it matches or undef if it does not match
+
+=cut
+
+sub db_version {
+    unless(pf::version::version_check()) {
+        add_problem ( $FATAL, "the database does not match the current version $PF_VERSION" );
```

Comment:
This does not block the httpd.admin from starting.
That is the only guarantee we can make in my opinion.